### PR TITLE
RQL request listener

### PR DESCRIPTION
**Pull request by hairmare**
This request listener looks for a configurable query string and then attempts to parse it.

The results of this parsing are then stored into the ``$request->attributes`` parameter bag where they can later be retrieved from.

This decouples parsing from using the query and helps me refactor RestBundle/Model/DocumentModel in graviton.

### CODE REVIEW

**Comment by narcoticfresh** [comment]
@hairmare Could you find out what exactly Symfony messes up in the parameter bag?

File: test/Tests/Listener/RequestListenerTest.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    /**
+     * @return array[]
+     */
+    public function willParseQueryData()
+    {
+        return [
+            'simple query string' => ['q=eq(foo,b%20a%20r)', 'eq(foo,b%20a%20r)'],
+            'string with paging stuff' => ['page=1&q=eq(foo,b%20a%20r)&perPage=20', 'eq(foo,b%20a%20r)'],
```
Comment (narcoticfresh) [inline test/Tests/Listener/RequestListenerTest.php]: @hairmare please include a complex URL case on a $ref here..

Comment (hairmare) [inline test/Tests/Listener/RequestListenerTest.php]: Good catch, added...

**Comment by narcoticfresh** [comment]
Travis is OK, let's not wait for scrutinizer..

**Pull request merged**